test/cmd/newapp.sh

```diff
@@ -167,4 +167,7 @@ os::cmd::expect_success 'oc new-app mongo -o json | python -m json.tool'
 # Ensure custom branch/ref works
 os::cmd::expect_success 'oc new-app https://github.com/openshift/ruby-hello-world#beta4'
 
+# Ensure the resulting BuildConfig doesn't have unexpected sources
+os::cmd::expect_success_and_not_text 'oc new-app https://github.com/openshift/ruby-hello-world -o=jsonpath="{.items[?(@.kind==\"BuildConfig\")].spec.source}"' 'dockerfile|binary'
```

Comment:
@bparees I am confident with the tests we added in #6615. If want to take another look:

https://github.com/openshift/origin/blob/38319a8338660479e1bfe6da31b6279865c5eba1/pkg/generate/app/cmd/newapp_test.go#L1131-L1165

and 

https://github.com/openshift/origin/blob/38319a8338660479e1bfe6da31b6279865c5eba1/pkg/generate/app/cmd/newapp_test.go#L1200-L1212


The comment about mocking was related to image discovery, it doesn't invalidate our existing tests, but just make certain scenarios hard to reproduce (with the current mocks, that we could improve and make it suitable for testing multiple match scenarios).